.github/workflows/lint-unit.yaml

```diff
@@ -0,0 +1,51 @@
+name: Linting and Unit Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+    branches:
+      - main
+
+jobs:
+  lint-unit:
+    name: Lint and Unit tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.6", "3.8", "3.10"]
```

Comment:
Since this job is going to execute 3 times (one for each python version), I think Sonarqube job would probably be triggered three time also. But I think it is acceptable because don't see a better alternative.

Reply:
I don't see how to get around it either